src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java

```diff
@@ -36,14 +36,11 @@ public SshCredentialDialog addCredential() {
 
         find(by.button("Add")).click();
 
-        if (getElement(by.xpath("//span[contains(@class,'credentials-add')]")) == null) {
+        String providerXpathExpr = "//div[contains(@class,'credentials-add-menu-items')]"
+                + "/div[@class='bd']/ul[@class='first-of-type']/li[contains(@class, 'yuimenuitem')]"
+                + "/span[contains(@class,'yuimenuitemlabel') and contains(@title, 'Jenkins Credentials Provider')]";
 
-            String providerXpathExpr = "//div[contains(@class,'credentials-add-menu-items')]"
-                    + "/div[@class='bd']/ul[@class='first-of-type']/li[contains(@class, 'yuimenuitem')]"
-                    + "/span[contains(@class,'yuimenuitemlabel') and contains(@title, 'Jenkins Credentials Provider')]";
-
-            find(by.xpath(providerXpathExpr)).click();
-        }
+        waitFor(by.xpath(providerXpathExpr)).click();
```

Comment:
switch to wait for so we know the "Jenkins Credentials Provider" button is visible before clicking it.

it would seem logical to me that the addition of new credentials should be part of a credentials page area but a quick scan did not show anything immediately obvious.